tests/test_conversion.py

```diff
@@ -0,0 +1,58 @@
+# License: MIT
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+"""Test conversion helper functions."""
+
+from datetime import datetime, timezone
+from math import isclose
+
+# pylint: disable=no-name-in-module
+from google.protobuf.timestamp_pb2 import Timestamp
+
+# pylint: enable=no-name-in-module
+from hypothesis import given
+from hypothesis import strategies as st
+
+from frequenz.client.base.conversion_helper import to_datetime, to_timestamp
+
+# Strategy for generating datetime objects
+datetime_strategy = st.datetimes(
+    min_value=datetime(1970, 1, 1),
+    max_value=datetime(9999, 12, 31),
+    timezones=st.just(timezone.utc),
+)
+
+# Strategy for generating Timestamp objects
+timestamp_strategy = st.builds(
+    Timestamp,
+    seconds=st.integers(
+        min_value=0,
+        max_value=int(datetime(9999, 12, 31, tzinfo=timezone.utc).timestamp()),
+    ),
+)
+
+
+@given(datetime_strategy)
+def test_to_timestamp_with_datetime(dt: datetime) -> None:
+    """Test conversion from datetime to Timestamp."""
+    ts = to_timestamp(dt)
+    assert ts is not None
+    converted_back_dt = to_datetime(ts)
+    # Compare timestamps to account for floating point errors
+    # Up to 1 second difference is allowed
+    assert isclose(dt.timestamp(), converted_back_dt.timestamp(), abs_tol=1)
```

Comment:
I really did find differences up to 1 second for certain timestamps. I suspect it's due to the floating point imprecision?

Reply:
Mmm, maybe it is because hypothosis uses huge values, and then the floating point rounding error translates to differences in the order of seconds. I wonder if we should limit testing to *reasonable values* (like years not bigger than `2_200` instead of 9999) to see if that gets a smaller error. If feels like allowing an imprecision of 1 second is a bit too much.

Reply:
It happens with `2000` as well...

```
>       assert isclose(dt.timestamp(), converted_back_dt.timestamp(), abs_tol=0.5)
E       assert False
E        +  where False = isclose(946684800.946685, 946684800.0, abs_tol=0.5)
E        +    where 946684800.946685 = <built-in method timestamp of datetime.datetime object at 0x7fde8f499680>()
E        +      where <built-in method timestamp of datetime.datetime object at 0x7fde8f499680> = datetime.datetime(2000, 1, 1, 0, 0, 0, 946685, tzinfo=datetime.timezone.utc).timestamp
E        +    and   946684800.0 = <built-in method timestamp of datetime.datetime object at 0x7fde8f498660>()
E        +      where <built-in method timestamp of datetime.datetime object at 0x7fde8f498660> = datetime.datetime(2000, 1, 1, 0, 0, tzinfo=datetime.timezone.utc).timestamp
E       Falsifying example: test_to_timestamp_with_datetime(
E           dt=datetime.datetime(2000, 1, 1, 0, 0, 0, 946685, tzinfo=datetime.timezone.utc),
E       )
```

Reply:
This made me check again and it seems I forgot micro/nano seconds from the converted timestamp. Pushed a fix.